aiida_quantumespresso/parsers/neb.py

```diff
@@ -95,168 +94,177 @@ def parse(self, **kwargs):
-            # avoiding repetitions.
-            if neb_out_dict['warnings']:
-                for warning in pw_out_data['warnings']:
-                    if warning not in neb_out_dict['warnings']:
-                        neb_out_dict['warnings'].append(warning)
+            # Add also PW warnings and errors to the neb output data, avoiding repetitions.
+            for log_type in ['warning', 'error']:
+                for message in logs_stdout[log_type]:
+                    formatted_message = '{}: {}'.format(log_type, message)
+                    if formatted_message not in neb_out_dict['warnings']:
+                        neb_out_dict['warnings'].append(formatted_message)
 
         # Symbols can be obtained simply from the last image
-        symbols = [ str(site.kind_name) for site in structure_data.sites]
-
-        new_nodes_list = []
+        symbols = [str(site.kind_name) for site in structure_data.sites]
 
-        # convert the dictionary into an AiiDA object
+        # convert output parameters to AiiDA Dict, and add as output node
         output_params = Dict(dict=dict(list(neb_out_dict.items())+list(image_data.items())))
+        self.out('output_parameters', output_params)
 
-        # return it to the execmanager
-        new_nodes_list.append((self.get_linkname_outparams(),output_params))
-
-        # convert data on structure of images into a TrajectoryData
+        # convert data on structure of images into a TrajectoryData, and add as output node
         traj = TrajectoryData()
         traj.set_trajectory(stepids = numpy.arange(1,num_images+1),
                             cells = numpy.array(cells),
                             symbols = numpy.array(symbols),
                             positions = numpy.array(positions),
                             )
+        self.out('output_trajectory', traj)
 
-        # return it to the execmanager
-        new_nodes_list.append((self.get_linkname_outtrajectory(),traj))
-
-        if parser_opts.get('all_iterations',False):
+        if parser_options.get('all_iterations', False):
             if iteration_data:
-                from aiida.orm.nodes.data.array import ArrayData
-
                 arraydata = ArrayData()
-                for x in six.iteritems(iteration_data):
-                    arraydata.set_array(x[0],numpy.array(x[1]))
-                new_nodes_list.append((self.get_linkname_iterationarray(),arraydata))
+                for k, v in six.iteritems(iteration_data):
+                    arraydata.set_array(k, numpy.array(v))
+                self.out('iteration_array', arraydata)
 
         # Load the original and interpolated energy profile along the minimum-energy path (mep)
         try:
-            mep_file = os.path.join( out_folder.get_abs_path('.'),
-                                                self._calc._PREFIX + '.dat' )
+            mep_file = os.path.join( out_folder._repository._get_base_folder().abspath,
```

Comment:
Use the `out_folder.open` method here. `loadtxt` should also accept filelike objects